### ISSUE TITLE
Slow loading time 1.7.2

### ISSUE THREAD

**Issue by trix0** [appium/appium#10070]
## The problem
It takes 35 seconds to init if all of the stuff is already in the device. If not it is around 1 minute. I was looking around a lot on appium and I found out that appium 1.6.3 starts the same code in 11 seconds if all the files are in phone already if not its 30seconds. It would be nice to have some flag that would allow us to get rid of unnecessary checks and  permissions and so on as in automated testing every second counts when you have 300 or more tests to run. 

https://discuss.appium.io/t/launching-the-app-became-slower-1-6-3-1-7-2/20580

## Environment

* Appium version (or git revision) that exhibits the issue:1.7.2
* Last Appium version that did not exhibit the issue (if applicable):1.6.3
* Desktop OS/version used to run Appium:Ubuntu 16.04
* Node.js version (unless using Appium.app|exe):9.4
* Mobile platform/version under test:Android
* Real device or emulator/simulator:Real Device
* Appium CLI or Appium.app|exe:CLI

## Details
I used no perms check which i thought could help but it didnt. 

## Link to Appium logs
https://gist.github.com/trix0/2bd3267fd6c6c820afed9e85c2733061      1.6.3
https://gist.github.com/trix0/bb405a695f77ce9a9a8fa6fd3ab8c293       1.7.2

## Code To Reproduce Issue [ Good To Have ]
https://gist.github.com/trix0/8588a7913922d7de62762debb8f71be1
Please remember that with sample code it's easier to reproduce the bug and it's much faster to fix it.

**Comment by trix0** [comment]
https://discuss.appium.io/t/launching-the-app-became-slower-1-6-3-1-7-2/20580

**Comment by mykola-mokhnach** [comment]
@trix0 Please try if https://github.com/appium/appium-android-driver/pull/317 works for you

**Comment by trix0** [comment]
yop

**Comment by trix0** [comment]
yo @mykola-mokhnach , What about put io.appium.settings to run on a background thread  and keep it running forever eve eve ever ? cause right  now it closes kinda often during my tests so this enhancement is good but quite not work always

**Comment by mykola-mokhnach** [comment]
We do put it into background. Perhaps, Android OS itself closes it after some time to free up resources. I'll anyway make this stuff better after https://github.com/appium/appium-adb/pull/290 is merged. Just waiting for an approval.

**Comment by trix0** [comment]
Lets hope IT ll pass. Appium is great just very very slow for testing at the current stage just because of this permission stuff. :-(

**Comment by trix0** [comment]
I used this answer https://stackoverflow.com/questions/34573109/how-to-make-an-android-app-to-always-run-in-background on io.appium.settings and it looks like that its stable now :) Maybe It ll help somebody. I do not think its for pull request

**Comment by trix0** [comment]
@mykola-mokhnach  Does not seem like #https://github.com/appium/appium-adb/pull/290 speeded up permissions , at least not on my side, does it work on  autoGrantPermissions:true or is it just used when it installs apk ?

**Comment by mykola-mokhnach** [comment]
These are just library methods. Now they need to be applied to the main drivers. Check https://github.com/appium/appium-android-driver/pull/319 for more details

**Comment by trix0** [comment]
Not sure what I am doing wrong in here... seems like   autoGrantPermissions:true is completely ignored.
I got yours latest repositories:
[adb](
https://github.com/appium/appium-adb/tree/41d63b25ce832503e9ef39830326fb8de2238462)
[android-driver](https://github.com/appium/appium-android-driver/tree/9fef577ec18ff421a2cd50c4aaef2fbc81516460)

Should I get something more ?

**Comment by mykola-mokhnach** [comment]
It's not done yet. I'll add it to this ticket when the changes are merged

**Comment by trix0** [comment]
Okey, Thanks If I can help you somehow just shoot me msg.

**Comment by mykola-mokhnach** [comment]
Thanks, it always good to have good support in testing.

**Comment by mykola-mokhnach** [comment]
@trix0 Please try the recent beta. android-driver component should be at 1.40+

**Comment by trix0** [comment]
@mykola-mokhnach https://github.com/appium/appium/releases/tag/v1.8.0-beta3 ? android driver is already 1.40 should i do somethin more ?

**Comment by trix0** [comment]
1.8 beta => loading time was 22sec. I believe its together with starting io.app.setting app. What is a drastic improvement(14 sec)
caps: https://gist.github.com/trix0/9968b41cd27125379b9b02c17e5366e3
log: https://gist.github.com/trix0/117c3a9f515b4b7ef54f7722fb5fbef8.

It seems that  autoGrantPermissions:true, still does not work tho.

**Comment by trix0** [comment]
I have actually another question there is a lot of checks.
AndroidDriver] Checking whether package is present on the device
2018-02-02 09:41:30:165 - info: [debug] [ADB] Running '/home/trixo/Android/Sdk/platform-tools/adb' with args: ["-P",5037,"-s","localhost:7409","shell","pm","list","packages","com.pointvoucher.playlondonpv"]

2018-02-02 09:41:31:816 - info: [debug] [ADB] Getting install status for io.appium.settings
2018-02-02 09:41:31:816 - info: [debug] [ADB] Getting connected devices...
2018-02-02 09:41:31:822 - info: [debug] [ADB] 2 device(s) connected
2018-02-02 09:41:31:823 - info: [debug] [ADB] Running '/home/trixo/Android/Sdk/platform-tools/adb' with args: ["-P",5037,"-s","localhost:7409","shell","pm","list","packages","io.appium.settings"]

and so on. 

It takes around 2 seconds to execute each of those commands so at the end its like 8 seconds of check which are not really necessary. Again 100 tests 8 seconds = 800 seconds.

Can we do something with that one as well ? 

Maybe something like desCap which would say: skipStrict:true/false ?

**Comment by mykola-mokhnach** [comment]
These checks are needed to compare the actual version code of the local apk and the code of the installed one. Do you know better way of doing that?

**Comment by trix0** [comment]
Well lets say, I have 10 tests. They run in parallel on 5 devices. 

First Test I have skipStric:false in desCaps. Therefore appium checks if everything is okey. If all apps that are supposed to be there are there with a correct version. 
Test Passed.

In Second test I change desCaps-> skipStrict to true, cause i already know(maybe full reset problem) that all the apps have the good version therefore I do not need checks.

What do you think about it. Like I can see why it might be bad idea to do that. It can result into unexpected errors in a production tests, but if everything is configured correctly it shouldnt. Not sure. I am not home in android nor adb therefore I have no idea about how  stuff works in there :) I wish I Could be more helpful but my web development skills are behind in this scenarios :-(

**Comment by mykola-mokhnach** [comment]
Create a separate feature request about this capability, so we can track it there. I close this issue as resolved for now.
